services/provider/server/consumer.go

```diff
@@ -224,3 +226,46 @@ func (c *ocsConsumerManager) Get(ctx context.Context, id string) (*ocsv1alpha1.S
 
 	return consumerObj, nil
 }
+
+func (c *ocsConsumerManager) UpdateStatusLastHeatbeat(ctx context.Context, id string) error {
+	uid := types.UID(id)
+
+	c.mutex.RLock()
+	consumerName, ok := c.nameByUID[uid]
+	if !ok {
+		c.mutex.RUnlock()
+		klog.Warningf("no storageConsumer found with UID %q", id)
+		return nil
+	}
+	c.mutex.RUnlock()
+
+	consumerObj := &ocsv1alpha1.StorageConsumer{}
+	consumerObj.Name = consumerName
+	consumerObj.Namespace = c.namespace
+
+	payload := []struct {
+		Op    string      `json:"op"`
+		Path  string      `json:"path"`
+		Value interface{} `json:"value"`
+	}{{
+		Op:    "replace",
+		Path:  "/status/lastHeartbeat",
+		Value: metav1.Now(),
+	}}
+	data, err := json.Marshal(payload)
```

Comment:
Let's simplify the anonymous struct to a single array item type, and rename the generic payload and data names to something more meaningful.  

```suggestion
	patchInfo  := struct {
		Op    string      `json:"op"`
		Path  string      `json:"path"`
		Value interface{} `json:"value"`
	} {
		Op:    "replace",
		Path:  "/status/lastHeartbeat",
		Value: metav1.Now(),
	}
	jsonPatchInfo, err := json.Marshal(interface[]{} {patchInfo})
```

Reply:
Updated in the latest commit